qiskit/pulse/schedule.py

```diff
@@ -230,14 +231,30 @@ def filter(self, *filter_funcs: List[Callable],
                channels: Optional[Iterable[Channel]] = None,
                instruction_types: Optional[Iterable[Type['Instruction']]] = None,
                time_ranges: Optional[Iterable[Tuple[int, int]]] = None,
-               intervals: Optional[Iterable[Interval]] = None) -> 'Schedule':
+               intervals: Optional[Iterable[Interval]] = None,
+               filter_type: Optional[int] = 0) -> Union['Schedule', Tuple['Schedule', 'Schedule']]:
         """
-        Return a new Schedule with only the instructions which pass though the provided filters.
-        Custom filters may be provided. If a list of channel indices is provided, only the
-        instructions that involve that channel (and maybe also others) will be included in the new
-        schedule. Similarly for instruction_types, only the instructions which are instances of the
-        provided types will be included. For intervals, instructions will be retained if their
-        timeslots are all wholly contained within *any* of the given intervals.
+        Filters the Schedule according to the specified filters, with behavior
+        depending on the filter_type argument (described below). Each filter is
+        a function taking (int, ScheduleComponent) and returning bool. The purpose is
+        to return a new Schedule containing only the parts of self that satisfy
+        a list of criteria.
+
+        filter_funcs are custom filters. If a list of channel indices is provided,
+        a filter returning True on those channels is created. Similarly, for
+        instruction_types, a filter returning True on those instruction types will
+        be created. For intervals, a filter returning True on instructions wholly
+        contained within *any* of the given intervals will be created.
+
+        The behaviour types:
+
+        If filter_type == 0:
+            returns a Schedule containing instructions returning True on all filters.
+        If filter_type == 1:
+            returns a Schedule containing the instructions excluded by filter_type==0
+        If filter_type == 2:
```

Comment:
Let's remove this option and only do "filter" and "not filter". As @dcmckayibm put in the issue, the goal is simply:
`schedule.filter(args) | schedule.exclude(args) == schedule`